tests/test_runtime.py

```diff
@@ -0,0 +1,119 @@
+import time
+from wandb.proto import wandb_internal_pb2 as pb
+
+
+def test_runtime(
+    internal_hm, mocked_run, mock_server, backend_interface, parse_ctx,
+):
+    with backend_interface() as interface:
+
+        proto_run = pb.RunRecord()
+        mocked_run._make_proto_run(proto_run)
+
+        run_start = pb.RunStartRequest()
+        run_start.run.CopyFrom(proto_run)
+
+        record = interface._make_request(run_start=run_start)
+        internal_hm.handle_request_run_start(record)
+
+        time.sleep(3)
+
+    ctx_util = parse_ctx(mock_server.ctx)
+
+    assert ctx_util.summary_wandb["runtime"] >= 3
+
+
+def test_runtime_pause_resume(
```

Comment:
[Optional]
Small nitpick here, but since this is a bit of duplicated code you could consolidate this by capturing the setup code all together.

You might consider using the `publish_util` fixture.  It was just extended to be more compatible with this type of pattern.

in your case you could either pass a function that does the RunStartRequest stuff as a begin_cb, and then your sleep and publish stuff as an end_cb, or you can make the RunStartRequest an optional part of publish_util fn so others could use it.